bindings/rust/s2n-tls-tokio/src/lib.rs

```diff
@@ -166,6 +166,16 @@ where
     C: AsRef<Connection> + AsMut<Connection> + Unpin,
     S: AsyncRead + AsyncWrite + Unpin,
 {
+    ///Access a reference to the underlaying io stream
+    pub fn io_ref(&self) -> &S {
```

Comment:
Would it be worth naming this the same as the `tokio-native-tls` crate: https://docs.rs/tokio-native-tls/latest/tokio_native_tls/native_tls/struct.TlsStream.html#method.get_ref

Might make it easier for folks to port from that.

Reply:
The issue is that we already implement AsRef<Connection> and AsMut<Connection> for `TlsStream`. Implementing or any new method with similar name will break the usage of 
```
stream.as_mut() / stream.as_ref() 
```
in various places where type cannot be inferred; as compiler will now expect type annotations.

alternative is to version bump s2n-tls-tokio and make the changes to support the similar API everywhere.

Reply:
They're not named the same thing, though. The `tokio-native-tls` crate uses `get_ref` and `get_mut`. I'm not saying that those names are better than `io_ref` and `io_mut` from a naming perspective either; just that it would make it one less thing to change for existing applications.

Reply:
Ah, I see. for some reason I had get_ref/as_ref equivalency in my head.

So, now the question is:

```
tlsStream.get_ref()->TcpStream
tlsStream.as_ref()->Connection
```

vs

```
tlsStream.io_ref()->TcpStream
tlsStream.as_ref()->Connection
```

Reply:
I don't have a strong preference here, but I think camshaft is right that we should lean towards matching other implementations and use get_ref(). mathpal, I think that was also your argument for adding the mut version when we talked earlier, and I think it applies to this choice as well.

Reply:
ack. I will make the change.